### PR TITLE
WIP: lpc176x5x: Fix SVD file

### PR DESCRIPTION
**Pull request by dbrgn**
I started fixing the lpc176x5x SVD file so that the generated code actually builds.

The duplicate keys were easy to fix. However, I'm stuck with errors like these:

```
error[E0428]: the name `MR` is defined multiple times
  --> src/pwm1.rs:92:1
   |
74 | pub struct MR {
   | ------------- previous definition of the type `MR` here
...
92 | pub struct MR {
   | ^^^^^^^^^^^^^ `MR` redefined here
   |
   = note: `MR` must be defined only once in the type namespace of this module
```

I believe it's due to this definition:

```xml
        <register>
            <dim>4</dim>
            <dimIncrement>0x4</dimIncrement>
            <dimIndex>0-3</dimIndex>
            <name>MR[%s]</name>
            <displayName>MR[%s]</displayName>
```

@rnestler you did a fix similar to this one for the lpc11uxx. Do you know how to fix this?

I also noticed that the description should probably contain a placeholder (MR%s) too:

```xml
<description>Match Register 0. MR0 can be enabled through the MCR to reset the TC, stop both the TC and PC, and/or generate an interrupt every time MR0 matches the TC.</description>
```

### CODE REVIEW

**Comment by rnestler** [comment]
> @rnestler you did a fix similar to this one for the lpc11uxx. Do you know how to fix this?

I'm not sure if it as this exact problem. But apparently the substitutions with `%s` do not work. Maybe a bug / missing feature in svd2rust?

**Comment by dbrgn** [comment]
Maybe @japaric @therealprof or @jamesmunns knows more about this?

**Comment by rnestler** [comment]
https://github.com/rust-embedded/svd2rust/issues/169 is probably related.

File: lpc176x5x/lpc176x5x.svd [hunk 1]
```diff
@@ -15662,12 +15661,12 @@ output edges.</description>
 					<enumeratedValues>
 					<name>ENUM</name>
 						<enumeratedValue>
-							<name>ERROR_OCCURRED_DURIN</name>
```
Comment (rnestler) [inline lpc176x5x/lpc176x5x.svd]: :smile: This looks like an error happened while generating the svd file :wink:

Comment (dbrgn) [reply lpc176x5x/lpc176x5x.svd]: No, look at the description:

> Error occurred during transmitting.

**Comment by dbrgn** [comment]
> rust-embedded/svd2rust#169 is probably related.

Yep, looks like that's the issue.

**Comment by dbrgn** [comment]
@rnestler uhm, why did you run the xmllint tool? Reformatting will make it much harder to diff against changes in case the upstream SVD changes... If you want to do it, it would be good to do it in a separate PR that we can discuss.

File: lpc176x5x/lpc176x5x.svd (hunk 1 above)
Comment (rnestler) [reply lpc176x5x/lpc176x5x.svd]: Yes, the joke was that there was *also* an error during generation because the value is truncated. :wink:

**Comment by rnestler** [comment]
> @rnestler uhm, why did you run the xmllint tool? Reformatting will make it much harder to diff against changes in case the upstream SVD changes... If you want to do it, it would be good to do it in a separate PR that we can discuss.

Hmm after thinking and discussing it offline I agree that it may be indeed bad to have the whitespace change to much. On the other hand we already changed the line-endings to unix, so one has to do at least these conversions to compare. Also the whitespace fixes in the descriptions I'd like to keep.

**Comment by dbrgn** [comment]
> On the other hand we already changed the line-endings to unix

    diff --strip-trailing-cr

:slightly_smiling_face: 

> Also the whitespace fixes in the descriptions I'd like to keep.

Yep, I agree about those!

File: lpc176x5x/lpc176x5x.svd (hunk 1 above)
Comment (dbrgn) [reply lpc176x5x/lpc176x5x.svd]: Ah :smile:

**Comment by hannobraun** [comment]
Closing due to inactivity.

@dbrgn Feel free to re-open, if you ever get a chance to update this.

**Pull request closed without merging**